oap-server/server-bootstrap/src/main/resources/otel-oc-rules/envoy.yaml

```diff
@@ -0,0 +1,40 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+expSuffix: tag({tags -> tags.app = "istio-dp::" + tags.app}).service(['app']).instance(['app'], ['kubernetes_pod_name'])
```

Comment:
Are you adding OpenTelemetry metrics for Envoy instances?

Reply:
I remember there was an `envoy.yaml` in envoy-metrics-rules from Envoy's native metrics service. What is the difference between these two?

Reply:
My plan was to replace the metrics in `envoy.yaml` with `otel-oc-rules/envoy.yaml`.

Because we begin to introduce Otel for Envoy metrics in this PR, I don't see any point to keep 2 ways of Envoy metrics.

Reply:
> My plan was to replace the metrics in `envoy.yaml` with `otel-oc-rules/envoy.yaml`.
> 
> Because we begin to introduce Otel for Envoy metrics in this PR, I don't see any point to keep 2 ways of Envoy metrics.

We don't need both for sure. But why we need to introduce OTEL? OTEL for Istio is what it provides natively, but Envoy has it natively. Does this new one provide new metrics? From the resource perspective, OTEL costs unnecessary resources.

Reply:
The metrics are essentially the same, but with the Envoy native metrics service, the formats are something like

```
cluster.outbound|9080|v2-mysql|ratings.default.svc.cluster.local.update_attempt
cluster.outbound|9080|v1|reviews.default.svc.cluster.local.update_attempt

cluster.outbound|9080|v2-mysql|ratings.default.svc.cluster.local.update_no_rebuild
cluster.outbound|9080|v1|reviews.default.svc.cluster.local.update_no_rebuild

cluster.outbound|9080|v2-mysql|ratings.default.svc.cluster.local.update_success
cluster.outbound|9080|v1|reviews.default.svc.cluster.local.update_success

cluster.outbound|9080|v2-mysql|ratings.default.svc.cluster.local.client_ssl_socket_factory.ssl_context_update_by_sds
cluster.outbound|9080|v1|reviews.default.svc.cluster.local.client_ssl_socket_factory.ssl_context_update_by_sds
```

and in Otel,

```
envoy_cluster_update_attempt{app=ratings,version=v2-mysql......}
envoy_cluster_update_attempt{app=reviews,version=v1......}
```

the latter one is more friendly to our MAL rules.

If we choose to use the Envoy native metrics service, we have to preprocess the metrics like Prometheus / Otel does

Reply:
According to our discussion, let's separate this into another PR.